src/AnnotationCreation.js

```diff
@@ -264,18 +276,26 @@ class AnnotationCreation extends Component {
                   aria-label="tool selection"
                   size="small"
                 >
-                  <ToggleButton value="rectangle" aria-label="add a rectangle">
-                    <RectangleIcon />
-                  </ToggleButton>
-                  <ToggleButton value="ellipse" aria-label="add a circle">
-                    <CircleIcon />
-                  </ToggleButton>
-                  <ToggleButton value="polygon" aria-label="add a polygon">
-                    <PolygonIcon />
-                  </ToggleButton>
-                  <ToggleButton value="freehand" aria-label="free hand polygon">
-                    <GestureIcon />
-                  </ToggleButton>
+                  { availableTools.indexOf('rectangle') !== -1 && (
+                    <ToggleButton value="rectangle" aria-label="add a rectangle">
+                      <RectangleIcon />
+                    </ToggleButton>
+                  )}
+                  { availableTools.indexOf('ellipse') !== -1 && (
+                    <ToggleButton value="ellipse" aria-label="add a circle">
+                      <CircleIcon />
+                    </ToggleButton>
+                  )}
+                  { availableTools.indexOf('polygon') !== -1 && (
+                    <ToggleButton value="polygon" aria-label="add a polygon">
+                      <PolygonIcon />
+                    </ToggleButton>
+                  )}
+                  { availableTools.indexOf('freehand') !== -1 && (
+                    <ToggleButton value="freehand" aria-label="free hand polygon">
+                      <GestureIcon />
+                    </ToggleButton>
+                  )}
```

Comment:
Since you are giving an ordered list of tools, it could also be rendered in the order of the availableTools list maybe, if you loop over availableTools and render based on the value.